src/lib.rs

```diff
@@ -132,6 +156,78 @@ impl<T: Transport + Sized> TapSigner<T> {
         }
         read_response
     }
+
+    pub fn certs_check(&mut self, nonce: Vec<u8>) -> Result<String, Error> {
+        let card_nonce = self.card_nonce.clone();
+
+        let certs_cmd = CertsCommand::default();
+        let certs_response: CertsResponse = self.transport.transmit(certs_cmd)?;
+
+        let check_cmd = CheckCommand::new(nonce.clone());
+        let check_response: Result<CheckResponse, Error> = self.transport.transmit(check_cmd);
+        if let Ok(response) = &check_response {
+            self.card_nonce = response.card_nonce.clone();
+        }
+
+        assert!(verify_tapsigner_signature(
+            &self.pubkey, 
+            check_response.unwrap().auth_sig.clone(), 
+            card_nonce, 
+            nonce.clone(), 
+            &self.secp
+        ).is_ok());
```

Comment:
Instead of an `assert!` how about adding new Error type `CertCheck(String)` and use it here like:
```suggestion
        ).map_err(|e| Error::CertCheck(e.to_string()))?;
```

Even though this is a fatal issue, we if just crash the app we don't have a chance to handle the error and tell the user what's going on.